server/auth/types/claims.go

```diff
@@ -93,19 +104,46 @@ func (c *Claims) GetUserInfoGroups(accessToken, issuer, userInfoPath string) ([]
 	request.Header.Set("Authorization", bearer)
 
 	response, err := httpClient.Do(request)
-
 	if err != nil {
 		return nil, err
 	}
-
-	userInfo := UserInfo{}
+	responseBody, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Could not read userInfo payload: %v", err)
+	}
+	log.Debug("UserInfo responseBody: ", string(responseBody))
 
 	defer response.Body.Close()
-	err = json.NewDecoder(response.Body).Decode(&userInfo)
 
+	userInfo := UserInfo{}
+	err = json.Unmarshal(responseBody, &userInfo)
 	if err != nil {
 		return nil, err
 	}
+	log.Debug("userInfo: ", userInfo)
+
+	// Default to getting groups using "groups" key
+	groups := userInfo.Groups
+
+	// If a custom group key was given, use that
+	if customKeyName != "" {
+		log.Info(
+			"A custom group key has been supplied, trying to retrieve the groups from key: ",
+			customKeyName,
+		)
+		var userInfoRawMap map[string]json.RawMessage
+
+		err = json.Unmarshal(responseBody, &userInfoRawMap)
+		if err != nil {
+			return nil, fmt.Errorf("Could not marshall userInfo payload")
```

Comment:
if we change the typings a bit, we can consolidate this by combining it with the top block, which I think would be a lot neater than doing 2 separate unmarshals.

then the conditional logic is just grabbing the custom claim